### PR TITLE
Code Challenge for Javier Searight

### PR DESCRIPTION
**Pull request by javiersearight**
@JRHonda 

I implemented the GetPersonQuery with the intention of getting the Person from the API, but there seemed to be an issue with that endpoint The error referenced the older REST API.

I left in the work I had done to demonstrate what I would have done if the API had worked as expected.

Thanks!

### CODE REVIEW

**Comment by javiersearight** [comment]
UPDATE:

As soon as I stepped away, I realized what the issues was with the GetPersonQuery

The endpoint supports two arguments for the 'id'. The reason I was getting the REST error was because the 'personID' argument was for the REST API and the 'id' argument was for the GraphQL support.

updating my PR.

File: app/src/main/java/com/lagunalabs/swapigraphql/MainActivity.kt [hunk 1]
```diff
@@ -45,71 +56,160 @@ import androidx.compose.ui.unit.sp
...
 
+    @OptIn(ExperimentalMaterial3Api::class)
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContent {
-            var showSheet by remember {
-                mutableStateOf(false)
+            var showSheet by remember { mutableStateOf(false) }
+
+            LaunchedEffect(true) {
```
Comment (JRHonda) [inline app/src/main/java/com/lagunalabs/swapigraphql/MainActivity.kt:73]: Code in this block will be executed asynchronously and only one time using `true` as the key.

File: app/src/main/java/com/lagunalabs/swapigraphql/MainActivity.kt [hunk 2]
```diff
@@ -45,71 +56,160 @@ import androidx.compose.ui.unit.sp
...
-
-                    NavHost(navController = navController, startDestination = "people") {
-                        composable("people") {
-                            InitPeople(onNavigateToPerson = {
-                                navController.navigate(
-                                    "person/{id}".replace(
-                                        oldValue = "{id}",
-                                        newValue = it
+                Surface {
+                    Scaffold(
```
Comment (JRHonda) [inline app/src/main/java/com/lagunalabs/swapigraphql/MainActivity.kt:85]: Use a scaffold to implement the different types of bars like the top bar, bottom bar, and snackbar.

File: app/src/main/java/com/lagunalabs/swapigraphql/MainActivity.kt [hunk 3]
```diff
@@ -45,71 +56,160 @@ import androidx.compose.ui.unit.sp
...
+                                        towards = AnimatedContentTransitionScope.SlideDirection.Companion.Right,
+                                        animationSpec = tween(400)
+                                    )
+                                },
+                                popExitTransition = {
+                                    slideOutOfContainer(
+                                        towards = AnimatedContentTransitionScope.SlideDirection.Companion.Right,
+                                        animationSpec = tween(400)
+                                    )
+                                }
```
Comment (JRHonda) [inline app/src/main/java/com/lagunalabs/swapigraphql/MainActivity.kt:141]: You can add transition animations to composables. These transitions make the screens slide left and right depending on whether or not the screen is being pushed to or away from.

File: app/src/main/java/com/lagunalabs/swapigraphql/MainActivity.kt [hunk 4]
```diff
@@ -122,58 +222,54 @@ class MainActivity : ComponentActivity() {
...
-                    colorResource(R.color.topColor),
-                    colorResource(R.color.bottomColor)
-                )
-        ) {
-            items(people.size) { index ->
-                people[index]?.apply {
-                    Row(
-                        horizontalArrangement = Arrangement.SpaceBetween,
+        LazyColumn {
+            itemsIndexed(items = people.mapNotNull { it }) { index, person ->
```
Comment (JRHonda) [inline app/src/main/java/com/lagunalabs/swapigraphql/MainActivity.kt:226]: can use this convenient method under `LazyListScope` to leverage index vs. passing in the List's size in `items(...)`

**Comment by JRHonda** [comment]
Hi Javier thank you for taking the time to do this homework assignment. I hope it was fair and helped you learn something. Our codebase is currently about 98% Jetpack Compose. I will discuss with the team and allow some of them to review your original submission commit, and we'll reach out.

**Comment by javiersearight** [comment]
Hey Justin,

Thanks for the feedback on my PR. This homework assignment was my first opportunity to use Compose in a practical way, and I actually found that I liked it.  There's a lot to learn but I'm liking how clean it is and the amount of ceremony that has been removed, especially around lists.

I've pulled down your commit and have been taking a deeper dive into it.

I especially appreciate the example of how to use Scaffold and TopAppBar. I looked into TopAppBar when I was working on the homework assignment, but backed away slowly because it was marked as Experimental.  I ran into Scaffold when I was researching the BottomSheet, but couldn't really understand what was the intended purpose. Your explanation was very helpful and helped set off a light bulb.

Happy Friday!

**Comment by JRHonda** [comment]
Thank you @javiersearight

**Pull request closed without merging**